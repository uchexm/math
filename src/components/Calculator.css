/* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}
*/

#root {
  display: grid;
  justify-content: center;
  align-items: center;
  min-height: 100vh;
}

.cal {
  display: inline-flex;
  justify-content: space-between;
  gap: 350px;
  margin: 0 0 100px 0;
}

h2 {
  display: contents;
}

.container {
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  height: 500px;
  width: 600px;
}

.num-input {
  padding: 15px 5px;
  grid-column: 1/5;
  text-align: end;
  background-color: #858293;
  outline: none;
  color: #fff;
}

#zero {
  grid-column: 1/3;
}

input {
  padding: 16px 25px;
  border-radius: initial;
  border: 1px solid rgb(218, 217, 217);
  cursor: pointer;
}

.input:hover {
  font-weight: 700;
  background-color: lightgray;
}

.operator {
  background-color: #ff8f0f;
  border: 1px solid whitesmoke;
}
