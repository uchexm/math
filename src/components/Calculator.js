import React from 'react';
import './Calculator.css';

/* eslint-disable*/
class Calculator extends React.PureComponent {
  render() {
    return (
      <div className="container">
        <Display />
        <Button value="AC" id="AC" />
        <Button value="+/-" id="sign" />
        <Button value="%" id="modulo" />
        <Button value="&divide;" id="divide" className="operator" />
        <Button value={7} id="seven" />
        <Button value={8} id="eight" />
        <Button value={9} id="nine" />
        <Button value="X" id="X" className="operator" />
        <Button value={4} id="four" />
        <Button value={5} id="five" />
        <Button value={6} id="six" />
        <Button value="-" id="minus" className="operator" />
        <Button value={1} id="one" />
        <Button value={2} id="two" />
        <Button value={3} id="three" />
        <Button value="+" id="plus" className="operator" />
        <Button value={0} id="zero" />
        <Button value="." id="dot" />
        <Button value="=" id="equal" className="operator" />
      </div>
    );
  }
}

const Button = ({ value, id, className }) => (
  <input type="button" value={value} id={id} className={className} />
);

const Display = () => <div className="num-input">0</div>;

export default Calculator;
