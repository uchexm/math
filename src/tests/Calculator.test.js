import React from 'react';
import { render } from '@testing-library/react';
import Calculator from '../components/Calculator';

describe('test for Calculator Component', () => {
  it('should render Calculator Component', () => {
    expect(render(<Calculator />)).toMatchSnapshot();
  });
});
