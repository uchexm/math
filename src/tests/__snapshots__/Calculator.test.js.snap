// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test for Calculator Component should render Calculator Component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="cal"
      >
        <div
          class="hd"
        >
          <h2>
            Let's do some Math!
          </h2>
        </div>
        <div
          class="container"
        >
          <div
            class="num-input"
          />
          <input
            id="AC"
            type="button"
            value="AC"
          />
          <input
            id="sign"
            type="button"
            value="+/-"
          />
          <input
            id="modulo"
            type="button"
            value="%"
          />
          <input
            class="operator"
            id="divide"
            type="button"
            value="รท"
          />
          <input
            id="seven"
            type="button"
            value="7"
          />
          <input
            id="eight"
            type="button"
            value="8"
          />
          <input
            id="nine"
            type="button"
            value="9"
          />
          <input
            class="operator"
            id="X"
            type="button"
            value="x"
          />
          <input
            id="four"
            type="button"
            value="4"
          />
          <input
            id="five"
            type="button"
            value="5"
          />
          <input
            id="six"
            type="button"
            value="6"
          />
          <input
            class="operator"
            id="minus"
            type="button"
            value="-"
          />
          <input
            id="one"
            type="button"
            value="1"
          />
          <input
            id="two"
            type="button"
            value="2"
          />
          <input
            id="three"
            type="button"
            value="3"
          />
          <input
            class="operator"
            id="plus"
            type="button"
            value="+"
          />
          <input
            id="zero"
            type="button"
            value="0"
          />
          <input
            id="dot"
            type="button"
            value="."
          />
          <input
            class="operator"
            id="equal"
            type="button"
            value="="
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="cal"
    >
      <div
        class="hd"
      >
        <h2>
          Let's do some Math!
        </h2>
      </div>
      <div
        class="container"
      >
        <div
          class="num-input"
        />
        <input
          id="AC"
          type="button"
          value="AC"
        />
        <input
          id="sign"
          type="button"
          value="+/-"
        />
        <input
          id="modulo"
          type="button"
          value="%"
        />
        <input
          class="operator"
          id="divide"
          type="button"
          value="รท"
        />
        <input
          id="seven"
          type="button"
          value="7"
        />
        <input
          id="eight"
          type="button"
          value="8"
        />
        <input
          id="nine"
          type="button"
          value="9"
        />
        <input
          class="operator"
          id="X"
          type="button"
          value="x"
        />
        <input
          id="four"
          type="button"
          value="4"
        />
        <input
          id="five"
          type="button"
          value="5"
        />
        <input
          id="six"
          type="button"
          value="6"
        />
        <input
          class="operator"
          id="minus"
          type="button"
          value="-"
        />
        <input
          id="one"
          type="button"
          value="1"
        />
        <input
          id="two"
          type="button"
          value="2"
        />
        <input
          id="three"
          type="button"
          value="3"
        />
        <input
          class="operator"
          id="plus"
          type="button"
          value="+"
        />
        <input
          id="zero"
          type="button"
          value="0"
        />
        <input
          id="dot"
          type="button"
          value="."
        />
        <input
          class="operator"
          id="equal"
          type="button"
          value="="
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
